packages/fdc3-get-agent/test/support/MockDocument.ts

```diff
@@ -6,24 +6,36 @@ export class MockDocument {
   name: string;
   window: MockWindow;
   iframes: MockIFrame[] = [];
+  static allDocuments: MockDocument[] = [];
 
   constructor(name: string, window: MockWindow) {
     this.name = name;
     this.window = window;
+    if (this.window.cw.debugLogs) {
+      console.log(`MockDocument (name: ${name} / window.name: ${this.window.name}): Created`);
+    }
+    MockDocument.allDocuments.push(this);
   }
 
   createElement(tag: string): HTMLElement {
     if (tag == 'iframe') {
-      const mw = new MockIFrame('iframe', this.window.cw, this.window, 'embedded-iframe');
+      if (this.window.cw.debugLogs) {
+        console.log(`MockDocument (name: ${this.name} / window.name: ${this.window.name}): creating iframe`);
+      }
+      const mw = new MockIFrame('iframe', this.window.cw, this.window, 'pending-iframe');
+      //n.b. variables are set on the parent window to link each iframe in order to allow use as source in
+      // postMessages from them - these are set in the MockIframe.setAttribute function
+
       this.iframes.push(mw);
-      return mw as any;
+
+      return mw as unknown as HTMLElement;
     } else {
-      return new MockElement(tag) as any;
+      return new MockElement(tag) as unknown as HTMLElement;
```

Comment:
It would be better if MockIFrame and MockElement were of type HTMLElement, so that you don't need to have `as unknown as HTMLElement` all over the place. Like, you might have to do it in the MockIFrame and MockElement class, but at least you'd only have to do it once.